packages/storage/src/providers/s3/types/results.ts

```diff
@@ -26,15 +34,20 @@ type S3ObjectInformation = {
 	 * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html#UserMetadata
 	 */
 	metadata?: Record<string, string>;
+	/**
+	 * VersionId used to reference a specific version of the object.
+	 */
+	versionId?: string;
 };
 
-export type S3DownloadDataResult = StorageDownloadDataResult &
-	S3ObjectInformation;
+export type S3DownloadDataResult = StorageDownloadDataResult & S3Item;
```

Comment:
I'd prefer to use generics like below:

```suggestion
export type S3DownloadDataResult = StorageDownloadDataResult<S3Item>;
```